### ISSUE TITLE
Music selector will not always open

### ISSUE THREAD

**Issue by thatactualdude** [chrislo27/RhythmHeavenRemixEditor#223]
### Information
**Log file**: 
[log_2017-12-13_19-59-45.txt](https://github.com/chrislo27/RhythmHeavenRemixEditor/files/1565273/log_2017-12-13_19-59-45.txt)

**Editor version**: (3.6.3)

### Description
When I did 5 remixes and made another one, I looked for music to use, then something wasn't right...

### Behaviours
**Expected behaviour**: I was supposed to see a selector ONLY REQUIRED for music files.

**Actual behaviour**: A selector didn't come up and I was stuck on that screen, nothing to do.

**Comment by chrislo27** [comment]
Missing log file.

Please be more specific with the actual behaviour. Did the screen not allow you to leave (did it explicitly say the chooser was still open, for example)?

**Comment by thatactualdude** [comment]
Yes, positive.

**Comment by chrislo27** [comment]
Please provide step by step instructions on how to reproduce this issue for yourself. They should always lead to a consistent and same problem.

**Comment by thatactualdude** [comment]
1. I click the 'select music' from the top bar
2. Wait for a few seconds.
3. Nothing comes up.

**Comment by chrislo27** [comment]
Please be more specific. For all I know, you could be right clicking the button and your instructions would lead to that. Also, "Nothing comes up" isn't exactly a step.

Does this happen ONLY with the music selector or does it apply to the save/load selectors as well?

**Comment by thatactualdude** [comment]
Save and load too.

**Comment by chrislo27** [comment]
Please update Java to the newest version (8u151). You have 8u141.

I would like to rule this out before assuming it is a program bug.

**Comment by thatactualdude** [comment]
Okay, will do.

**Comment by thatactualdude** [comment]
Updated, but it is still happening.

**Comment by thatactualdude** [comment]
Here is a new log after updating:
[log_2017-12-16_19-12-20.txt](https://github.com/chrislo27/RhythmHeavenRemixEditor/files/1565285/log_2017-12-16_19-12-20.txt)

**Comment by chrislo27** [comment]
I have just tested it for a considerable amount of time on El Capitan. All file dialogs work (open, save, export, music).

I do not know what is wrong but it could be OS dependent for all I know since High Sierra is relatively new.

**Comment by thatactualdude** [comment]
This may be kind of off-topic, but will there be another release that will NOT have this problem in High Sierra?

**Comment by chrislo27** [comment]
I don't know. So far you are the only one to have reported this issue with High Sierra.

Again, I didn't say it was absolutely OS dependent, but it's a possibility BECAUSE I tested with El Capitan. There are a lot of settings and other unknown variables that you could have that others don't. The log files also tell me nothing related to the file selectors.

**Comment by thatactualdude** [comment]
Okay, so I've tried reinstalling RHRE, but it's still happening. Is it possible if I can use an older version?

**Comment by chrislo27** [comment]
Be my guest. If you can find the point between versions at which this doesn't occur I'll gladly look and see what changes were made between then.

A reminder that everything you have said and provided so far does not lead to anything that tells me this is happening.

**Comment by thatactualdude** [comment]
I've downloaded from 3.6.3 to 3.5.0, but nothing is working!

**Comment by chrislo27** [comment]
v3.6.3 to v3.5.0 is only six versions (counting every patch) up to late October. v3.0.4 dates up to late August. There is quite a large time gap between August and October. I recommend testing v3.3.0, and v3.2.2, as file choosers had only one significant change between the two versions.

You should also clarify something: in the first post you say that you made five remixes and THEN the selector doesn't open. **Is this still the case, or does the selector now NEVER open?**

If we fail to discover the root cause of this problem in the next 48 hours I will rule this as an unreproducible bug and close the issue. Failing to be specific with details only leads to dragging on issues for long periods of time, and issues expire.

**Comment by thatactualdude** [comment]
Well, after clicking the selectors for multiple times, I'm thinking it'll never open from now on.

**Comment by chrislo27** [comment]
Some other things you may want to try:
* Reboot the computer
  * Supposedly JavaFX natives may have issues between application starts, not to mention High Sierra being more recent than El Capitan
* Launch with the JVM arguments `-Dprism.verbose=true -Dprism.debug=true` and provide the log file
  * To do this, open Finder, right click the folder where the jar file is, click Open Terminal at Folder (or something similar), and type: `java -jar -Dprism.verbose=true -Dprism.debug=true RHRE3.jar`
  * Provide the log file for this run (while trying to open the file choosers, of course) afterwards because the two arguments may output more important details

**Comment by thatactualdude** [comment]
I've already rebooted before, but that didn't work.
I've also tried the second method, but I was stuck on this screen:
![screen shot 2017-12-20 at 6 41 13 pm](https://user-images.githubusercontent.com/34498702/34238235-7a433f6a-e5b5-11e7-9d72-19802ef31250.png)
Also, here's what happened in the terminal during choosing the music selector:


00:00:02.491: [INFO][LWJGL Application] Resizing camera as KEEP_ASPECT_RATIO, window is 1280 x 720, camera is 1280.0 x 720.0
00:00:03.537: [INFO][LWJGL Application] RHRE3 v3.6.3 is starting...
00:00:03.599: [INFO][LWJGL Application] Running on JRE 1.8.0_102 (is at least 9?: false)
JavaSoundAudioIO: Chosen mixer is Default Audio Device.
JavaSoundAudioIO: Chosen mixer is Default Audio Device.
00:00:04.724: [INFO][LWJGL Application] Found 0 json palette files, attempting to read
00:00:06.530: [INFO][LWJGL Application] Resizing camera as KEEP_ASPECT_RATIO, window is 1280 x 720, camera is 1280.0 x 720.0
Prism pipeline init order: es2 sw 
Using native-based Pisces rasterizer
Using dirty region optimizations
Not using texture mask for primitives
Not forcing power of 2 sizes for textures
Using hardware CLAMP_TO_ZERO mode
Opting in for HiDPI pixel scaling
Prism pipeline name = com.sun.prism.es2.ES2Pipeline
Loading ES2 native library ... prism_es2
	succeeded.
GLFactory using com.sun.prism.es2.MacGLFactory
(X) Got class = class com.sun.prism.es2.ES2Pipeline
Initialized prism pipeline: com.sun.prism.es2.ES2Pipeline
00:00:09.083: [INFO][LWJGL Application] Resizing camera as KEEP_ASPECT_RATIO, window is 1920 x 1080, camera is 1920.0 x 1080.0
00:00:09.320: [INFO][LWJGL Application] Reloaded all 5 fonts in 236.876996 ms
Maximum supported texture size: 16384
Maximum texture size clamped to 4096
Non power of two texture support = true
Maximum number of vertex attributes = 16
Maximum number of uniform vertex components = 4096
Maximum number of uniform fragment components = 4096
Maximum number of varying components = 60
Maximum number of texture units usable in a vertex shader = 16
Maximum number of texture units usable in a fragment shader = 16
Graphics Vendor: Intel Inc.
       Renderer: Intel HD Graphics 5000 OpenGL Engine
        Version: 2.1 INTEL-10.28.29
 vsync: true vpipe: true
00:00:11.987: [INFO][ForkJoinPool.commonPool-worker-2] Pulled GitHub version in 2984.7034 ms, got 29 vs real 29
00:00:11.987: [INFO][ForkJoinPool.commonPool-worker-1] Fetched editor version from GitHub in 5457.391 ms, is v3.6.3
00:00:17.871: [INFO][ForkJoinPool.commonPool-worker-2] Fetching...
00:00:18.729: [INFO][ForkJoinPool.commonPool-worker-2] Resetting...
00:00:22.037: [INFO][ForkJoinPool.commonPool-worker-2] Finished fetch/clone in 13034.565906 ms
QuantumRenderer: shutdown
